### DISCUSSION TITLE
Using regex and rewrite in nginx location when proxying Plausible

### DISCUSSION THREAD

**Discussion by mheland** [plausible/analytics#2451] [Anything Else]
Proxying self-hosted Plausible on Digital Ocean VPS to bypass adblockers.
Wanted to enable outbound link tracking and had to change the script name in nginx config to script.outbound-links.js

After some googling I learned that I could use a regex in `location` and proxy everything under /js/ 

nginx can also rewrite the script name (outbound-links.js is blocker-bait) so now I have 

```
 location ^~ /js/ {                                            # regex, notice trailing slash

     rewrite ^/js/pl.js$ /js/script.js last;                   # rewrite scriptjs to pl.js
     rewrite ^/js/pl2.js$ /js/script.outbound-links.js last;   # rewrite script.outbound-links.js to pl2.js

     proxy_pass https://myplausiblehost.com/js/;               # notice trailing slash

```

The script tag now flies under any ad-blocking radar with...
`<script defer data-api="/api/event" data-domain="mywebsite.com" src="/js/pl2.js"></script>`

Thank you to very many people who shared on Stack Overflow and to amazing developers at Plausible.

**Comment by ukutaht** [comment]
Nice, thanks for sharing @mheland !
